### PR TITLE
Backports redirects from #12528, adds indexing page

### PR DESCRIPTION
**Pull request by C-STYR**
- backports redirects from #12528 
- adds indexing page for future sunsetting of older docs versions
- adds redirect for broken link in Teleport App

### CODE REVIEW

**Review by jimbishopp — APPROVED** [review]

File: docs/pages/older-versions.mdx
Comment (ptgott) [inline docs/pages/older-versions.mdx:1]: I think we should forward port this page to `master` and backport it to `branch/v8` and `branch/v7` as well so users of all supported versions can view this page.

Also, just double-checking whether the intention is to make this page accessible only to visitors who know the URL, or if there will be a link from a layout in `gravitational/docs`.

File: docs/pages/older-versions.mdx
Comment (ptgott) [inline docs/pages/older-versions.mdx:2]:
```suggestion
title: Older Versions of the Teleport Documentation
```

Comment (ptgott) [inline docs/pages/older-versions.mdx:2]:
```suggestion
title: Older Versions of Teleport the Documentation
```
```suggestion
title: Older Versions of Teleport Documentation
```

File: docs/pages/older-versions.mdx
Comment (ptgott) [inline docs/pages/older-versions.mdx:3]:
```suggestion
description: Links to older versions of the Teleport docs.
```

Comment (ptgott) [inline docs/pages/older-versions.mdx:3]:
```suggestion
description: Links to older versions of the Teleport docs.
```
```suggestion
description: Links to older versions of the Teleport Docs
```

File: docs/pages/older-versions.mdx
Comment (ptgott) [inline docs/pages/older-versions.mdx:7]:
```suggestion
```
Since the title will already appear at the top of the page

Comment (ptgott) [inline docs/pages/older-versions.mdx:7]:
```suggestion
```
Since the title will already appear at the top of the page

File: docs/pages/older-versions.mdx
Comment (ptgott) [inline docs/pages/older-versions.mdx:11]: Would this list look nicer as a `TileSet`/`Tile`s?

Comment (ptgott) [inline docs/pages/older-versions.mdx:11]: Would this list look nicer as a `TileSet`/`Tile`s?

File: docs/pages/older-versions.mdx
Comment (ptgott) [inline docs/pages/older-versions.mdx:4]:
```suggestion
```

I think we can use the title here instead of providing a separate H1.

Comment (ptgott) [inline docs/pages/older-versions.mdx:4]:
```suggestion
```

I think we can use the title here instead of providing a separate H1.

Comment (C-STYR) [reply docs/pages/older-versions.mdx:4]: All pages need an H1 for SEO purposes @ptgott

File: docs/pages/older-versions.mdx
Comment (C-STYR) [reply docs/pages/older-versions.mdx:10]: You're welcome to style it however you like, for now this is just a placeholder for testing purposes.

File: docs/pages/older-versions.mdx
Comment (C-STYR) [reply docs/pages/older-versions.mdx:1]: Once I get the necessary approvals for this test page it will be included in `master`, but for now I don't want to have to update multiple PRs.  It does not need to be backported - that will be handled during redirection.
Many roads and regex matches will lead here eventually, and there will be a link from the versions dropdown.

**Review by r0mant — APPROVED** [review]

**Pull request merged**
